### PR TITLE
Update top-level README

### PR DESCRIPTION
**Pull request by bernardnormier**
This PR reworks the main README.md.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,112 +1,115 @@
...
+    {
+        auto ipInfoPtr = dynamic_pointer_cast<Ice::IPConnectionInfo>(
+            current.con->getInfo());
+
+        if (ipInfoPtr)
+        {
+            cout << "Received hello from "
+                 << ipInfoPtr->remoteAddress << ":" << ipInfoPtr->remotePort
+                 << endl;
+        }
```
Comment (bentoi) [inline README.md:63]: I would just print "Hello world!" instead of getting the IP/port which makes the code more complicated.

**Review by bentoi — APPROVED** [review]
Looks good!

**Review by pepone — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (ReeceHumphreys) [reply README.md:63]: Yes, I think making it a bit simpler would be good

**Review by ReeceHumphreys — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -1,112 +1,115 @@
...
+```
+
+You can use any supported programming language for your client and server. For example, a Python client could call a C++
+server; neither side knows the programming language used by the other side.
+
+## Complete solution with a uniform API
+
+The Ice framework provides everything you need to build networked applications:
+- RPCs with a compact binary protocol over a variety of network transports (TCP, UDP, WebSocket, Bluetooth...)
+- Secure communications (IceSSL)
```
Comment (externl) [inline README.md:75]: It would be nice to make these doc links

**Review by externl — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (bernardnormier) [reply README.md]: Added links to the documentation.

**Pull request merged**